.github/workflows/release.yml

```diff
@@ -135,44 +132,36 @@ jobs:
                   name: ark-${{ matrix.flavor }}-linux-arm64-archive
 
             - name: Upload macOS release artifact (universal)
-              uses: actions/upload-release-asset@v1
+              uses: softprops/action-gh-release@v2
               env:
                   GITHUB_TOKEN: ${{ github.token }}
               with:
-                  upload_url: ${{ needs.create_release.outputs.upload_url }}
-                  asset_path: ark-${{ needs.get_version.outputs.ARK_VERSION }}${{ env.DEBUG_FLAG }}-darwin-universal.zip
-                  asset_name: ark-${{ needs.get_version.outputs.ARK_VERSION }}${{ env.DEBUG_FLAG }}-darwin-universal.zip
-                  asset_content_type: application/octet-stream
+                  tag_name: ${{ needs.get_version.outputs.ARK_VERSION }}
+                  files: ark-${{ needs.get_version.outputs.ARK_VERSION }}${{ env.DEBUG_FLAG }}-darwin-universal.zip
 
             - name: Upload Windows release artifact (x64)
-              uses: actions/upload-release-asset@v1
+              uses: softprops/action-gh-release@v2
               env:
                   GITHUB_TOKEN: ${{ github.token }}
               with:
-                  upload_url: ${{ needs.create_release.outputs.upload_url }}
-                  asset_path: ark-${{ needs.get_version.outputs.ARK_VERSION }}-${{ matrix.flavor }}-windows-x64.zip
-                  asset_name: ark-${{ needs.get_version.outputs.ARK_VERSION }}${{ env.DEBUG_FLAG }}-windows-x64.zip
-                  asset_content_type: application/octet-stream
+                  tag_name: ${{ needs.get_version.outputs.ARK_VERSION }}
+                  files: ark-${{ needs.get_version.outputs.ARK_VERSION }}${{ env.DEBUG_FLAG }}-windows-x64.zip
```

Comment:
The main difficulty is that this new softprops action does not distinguish between `asset_path` (where it came from) and `asset_name` (what you want to upload as).

Previously for Windows and Linux we _slightly_ tweaked the `asset_path` name to remove `-release` for release builds. That's the `env.DEBUG_FLAG` usage above. Note that on macOS, we were already uploading the asset at `asset_path` with `-release` removed, so no tweaking was being done there.

An easy solution is to simply upload the intermediate zips in `release-windows.yml` and `release-linux.yml` with the finalized names that we want to upload to the actual release - i.e. we follow what `release-macos.yml` was already doing and go ahead and remove the `-release` from the name when it isn't a debug build.